### PR TITLE
[issue-5415] Added support for XML example generation

### PR DESCRIPTION
**Pull request by joymarielim**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Added support for support for XML example generation

### CODE REVIEW

**Comment by joymarielim** [comment]
@wing328, here's the PR that would add support for XML example generation on Spring apps.
CC: @elizabetht

**Comment by wing328** [comment]
@joymarielim would you please rebase on the latest master to resolve merge conflicts?

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/examples/XmlExampleGenerator.java [hunk 1]
```diff
@@ -231,4 +217,17 @@ protected String indent(int indent) {
         }
         return sb.toString();
     }
+
+    private String compressTagWhenEmpty(String name, String xmlNode){
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/examples/XmlExampleGenerator.java]: @joymarielim would be good to add a one-liner explaining `compressTagWhenEmpty`

File: modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache [hunk 2]
```diff
@@ -77,5 +78,12 @@ public class {{classname}}Controller implements {{classname}} {
 
 {{/operation}}
 {{/jdk8-no-delegate}}
+    private ObjectMapper getObjectMapperByContentType(String mediaType){
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache]: @joymarielim would be good to add a docstring to explain what this function does

**Comment by wing328** [comment]
When you've time, please rebase on the latest master to resolve the merge conflicts.

**Comment by joymarielim** [comment]
@wing328 , I've addressed the review points and added comments accordingly.
CC: @elizabetht

**Comment by elizabetht** [comment]
@joymarielim We have to move this change to 2.3.0

**Comment by wing328** [comment]
As @elizabetht pointed out, please resubmit this PR for 2.3.0 branch instead.

**Pull request closed without merging**